### PR TITLE
test: Retry activating a the new tun interface a couple of times

### PR DESCRIPTION
**Pull request by mvollmer**
Because that's what you do.

### CODE REVIEW

**Comment by stefwalter** [comment]
Is this Debian specific? Is there an upstream bug report for this?

File: test/verify/check-networking [hunk 1]
```diff
@@ -419,7 +419,7 @@ class TestNetworking(MachineCase):
 
         # Create a tun device and let NetworkManager manage it
         m.execute("ip tuntap add mode tun dev %s" % iface)
-        m.execute("nmcli --wait 0 dev con %s" % iface)
+        wait(lambda: m.execute("nmcli --wait 0 dev con %s" % iface) == "")
```
Comment (stefwalter) [inline test/verify/check-networking]: Why do we have --wait 0 here? Am I misunderstanding what that does?

Comment (mvollmer) [inline test/verify/check-networking]: The activation will not succeed fully.  The interface will not receive a IP address, for example.  We just want NM to recognize this interface and mark it as "managed".

**Comment by mvollmer** [comment]
> Is this Debian specific? Is there an upstream bug report for this?

I don't know enough about what is really happening.  I file a bug and ask for help.

**Comment by mvollmer** [comment]
>  I file a bug and ask for help.

https://bugzilla.gnome.org/show_bug.cgi?id=774266

From IRC:

```
<mvollmer> so, "ip tuntap add mode tun dev tun0; nmcli --wait 0 dev con tun0" mostly works, but sometimes fails
<mvollmer> the error is... wait a second
<mvollmer> Failed to add/activate new connection: Connection 'tun0' is not available on the device tun0 at this time.
<mvollmer> could there be a race, where the NM daemon hasn't caught up with udev events or similar?
<mvollmer> note that the error is not "No such device"
<lubko> yes, probably
<mvollmer> modern linux is full of races
<mvollmer> lubko, does it make sense to report this as a bug?
<lubko> mvollmer: yes
<mvollmer> I can't recall why I used "ip tuntap add" and didn't let NM do the whole job...
<lubko> mvollmer: we should indeed have something like "nmcli settle" that would block until NetworkManager catches up with udev/rtnetlink
<lubko> if we decide it makes sense to externally create the device and connect it with NM of course
```

**Comment by stefwalter** [comment]
Nice. I'll put that link in as I merge.

**Pull request closed without merging**